### ISSUE TITLE
complex encoding problems

### ISSUE THREAD

**Issue by Levitanus** [RomeoDespres/reapy#73]
Well, it is not just about `reapy`, but about the `reaper_python` itself also.
At first, `reaper_python.py` uses `encode('utf-8')` for the any of string arguments taken to and from the C interface. But, here the thing, if we take, for example the `RPR_MIDI_GetEvnt` with SysEx, for example, it will crash as it cannot be encoded to the utf-8.

Moving forward I've patched the function just to see where it can lead to, and, occasionally, reapy can't work with bytes also, so, I've tried to patch reapy server, and it, somehow, still tries to dump the bytes. I suppose to can handle it at the next iterations, but here the place I stopped, as now I feel that do some stuff in `lua` is cheaper, than do on python.

So, I have a question: what do we do? Look for all the "bytes" functions in the original `reaper_python.py` and ask Justin to fix 'em or make a lua helper for the particular things? I ask, as I never succeed with bug reports on the cokos forum.

**Comment by Levitanus** [comment]
https://forum.cockos.com/showthread.php?t=234475

**Comment by Levitanus** [comment]
Romeo, while you are here, may you look to the original `reaper_python.py`, suggesting the changes? Last time I was too confused about all the encodings and moved big part of the code to the lua.

**Comment by Levitanus** [comment]
Ooops 🧐
As an Option, we can fix it by ourselves:

```Python
import ctypes as ct
from reaper_python import (
    _ft, rpr_initft, rpr_getfp, rpr_packp, rpr_unpackp, rpr_packsc, rpr_packs,
    rpr_unpacks
)
import reapy as rpr


def ShowConsoleMsg(p0: str) -> None:
    a = _ft['ShowConsoleMsg']
    f = ct.CFUNCTYPE(None, ct.c_char_p)(a)
    t = (rpr_packsc(p0), )
    f(t[0])


ShowConsoleMsg('hello world')


def JS_Window_OnCommand(windowHWND: int, commandID: int) -> bool:
    a = _ft['JS_Window_OnCommand']
    f = ct.CFUNCTYPE(ct.c_bool, ct.c_int, ct.c_int)(a)
    t = (rpr_packp('HWND', windowHWND), ct.c_int(commandID))
    return f(t[0], t[1])


c_id = 7
hwnd = rpr.reascript_api.GetMainHwnd()
JS_Window_OnCommand(hwnd, c_id)
```

**Comment by Levitanus** [comment]
fixed in #74
